### PR TITLE
Aquila H32 Fixes / Merge w/Marlin current bugfix-2.1.x

### PR DESCRIPTION
**Pull request by classicrocker883**
### Description

Update to most recent bugfix,  
more importantly, some fixes to be addressed.

### CODE REVIEW

**Comment by shadow578** [comment]
thanks for the work!

did you have to do any changes in this commit [6a88e16](https://github.com/shadow578/Marlin-H32/pull/49/commits/6a88e16fb9b856785564109186ebc29153c12f90)? or is it just git being git?

File: Marlin/src/lcd/e3v2/proui/dwin.cpp
Comment (shadow578) [inline Marlin/src/lcd/e3v2/proui/dwin.cpp:1]: is it correct that this change is from [9d1d901](https://github.com/shadow578/Marlin-H32/pull/49/commits/9d1d901a0852cbd42c1c88144a890e793ba37c20)?

File: platformio.ini [hunk 1]
```diff
@@ -13,7 +13,7 @@
 [platformio]
 src_dir      = Marlin
 boards_dir   = buildroot/share/PlatformIO/boards
-default_envs = mega2560
+default_envs = HC32F460xCxx_AQUILA_V101
```
Comment (shadow578) [inline platformio.ini]:
```suggestion
default_envs = mega2560
```

i'd like to keep the marlin source code as close to upstream as possible, so please revert this change

**Review by shadow578 — CHANGES_REQUESTED** [review]
assuming all changes outside of `Marlin/src/HAL/HC32F46x` are from merged commits.

see comments, otherwise LGTM

File: Marlin/src/HAL/HC32F46x/dogm/u8g_com_stm32duino_swspi.cpp [hunk 2]
```diff
@@ -35,7 +34,6 @@ static uint8_t SPI_speed = SPI_SPEED;
 
 static inline uint8_t swSpiTransfer_mode_0(uint8_t b, const uint8_t spi_speed, const pin_t miso_pin = -1)
 {
-  for(int i = 0; i < 8; i++)
```
Comment (shadow578) [inline Marlin/src/HAL/HC32F46x/dogm/u8g_com_stm32duino_swspi.cpp:38]: i assume this is related to https://github.com/shadow578/Marlin-H32/commit/50a9fab5b477c6906c2e88c4bc157fdf30d9914f#r130459559?

File: Marlin/src/lcd/e3v2/proui/dwin.cpp
Comment (classicrocker883) [reply Marlin/src/lcd/e3v2/proui/dwin.cpp:1]: I wrote the original code and did a [PR] to Marlin, this change should be pulled to the original soon. so yes, any changes from that commit are additional.

File: Marlin/src/HAL/HC32F46x/dogm/u8g_com_stm32duino_swspi.cpp (hunk 2 above)
Comment (classicrocker883) [reply Marlin/src/HAL/HC32F46x/dogm/u8g_com_stm32duino_swspi.cpp:38]: yes exactly, I wasn't sure if I was going to do a [PR], so that previous comment can be ignored.

**Review by classicrocker883 — COMMENTED** [review]
I do not know if pulling from the Marlin origin was helpful or not. (after posting this [PR])  
I'm sure you have your way so things don't get mixed up. let me know if pulling from the current Marlin _bugfix-2.1.x_ should be reverted or if it's fine leaving as it is.

**Comment by classicrocker883** [comment]
> thanks for the work!
> 
> did you have to do any changes in this commit [6a88e16](https://github.com/shadow578/Marlin-H32/pull/49/commits/6a88e16fb9b856785564109186ebc29153c12f90)? or is it just git being git?

There should not be any of my changes in that commit. what I propose [should be here - `fixes for H32`](https://github.com/shadow578/Marlin-H32/pull/49/commits/9d1d901a0852cbd42c1c88144a890e793ba37c20)


If you haven't know how to do so, I simply added the remote to this repo and upstream to MarlinFirmware, and then did a Pull from...
